src/core/thread/mle_router.cpp

```diff
@@ -4857,6 +4858,30 @@ otError MleRouter::GetMaxChildTimeout(uint32_t &aTimeout) const
     return error;
 }
 
+void MleRouter::SignalChildUpdated(otThreadChildTableEvent aEvent, Child &aChild)
+{
+    otChildInfo childInfo;
+    otError error;
+
+    error = GetChildInfo(aChild, childInfo);
+
+    if ((mChildTableChangedCallback != NULL) && (error == OT_ERROR_NONE))
```

Comment:
Is there a valid use case for `error != OT_ERROR_NONE`?  If not, should we assert instead?

Reply:
Good point! The `child` should be in `valid` state when this is called and therefore `GetChildInfo` should not fail.
Will change it to assert.